backend/regalloc/regalloc_utils.ml

```diff
@@ -611,22 +510,30 @@ let insert_block :
     Cfg_with_layout.t ->
     Cfg.basic_instruction_list ->
     after:Cfg.basic_block ->
+    before:Cfg.basic_block option ->
     next_instruction_id:(unit -> Instruction.id) ->
-    unit =
- fun cfg_with_layout body ~after:predecessor_block ~next_instruction_id ->
+    Cfg.basic_block list =
+ fun cfg_with_layout body ~after:predecessor_block ~before:only_successor
+     ~next_instruction_id ->
   let cfg = Cfg_with_layout.cfg cfg_with_layout in
   let successors =
-    Cfg.successor_labels ~normal:true ~exn:false predecessor_block
+    match only_successor with
+    | None -> Cfg.successor_labels ~normal:true ~exn:false predecessor_block
+    | Some only_successor -> Label.Set.singleton only_successor.start
   in
   if Label.Set.cardinal successors = 0
   then
     Misc.fatal_errorf
       "Cannot insert a block after block %a: it has no successors" Label.print
       predecessor_block.start;
-  let last_insn =
+  let dbg, fdo, live, stack_offset =
     match DLL.last body with
-    | None -> Misc.fatal_error "Inserting an empty block"
-    | Some i -> i
+    | None ->
```

Comment:
I think it can happen in a subsequent pull request, although that could be detected an optimised. Can we leave a CR or a comment on 758 to revisit when the chain is merged?

Reply:
> would empty liveness always be recomputed?

Yes, liveness will be set by [`postlude`](https://github.com/ocaml-flambda/flambda-backend/blob/main/backend/regalloc/regalloc_rewrite.ml#L229), at the end
of register allocation.

Reply:
> I think it can happen in a subsequent pull request, although that could be detected an optimised. Can we leave a CR or a comment on 758 to revisit when the chain is merged?

sounds good, a comment on 758 is enough, thanks.